packages/analytics-core/src/timeline.ts

```diff
@@ -106,7 +113,7 @@ export class Timeline {
     );
 
     const executeDestinations = destination.map((plugin) => {
-      return plugin.flush && plugin.flush();
+      return plugin.flush && plugin.flush(true);
```

Comment:
Add the `flush(true)` here as the logic in `Destination` actually use `useRetry` here: https://github.com/amplitude/Amplitude-TypeScript/blob/main/packages/analytics-core/src/plugins/destination.ts#L129. The response might be slightly different.

Reply:
🤔 this changes the behavior of `client.flush()` since that solves flushing events immediately before an app closes. I'm worried that if retry, for node it will wait for the promises to resolve/reject before the app closes. I think in client, instead of calling flush - we might need to call scheduleApply instead. What do you think?